### PR TITLE
[IOTDB-6187] IoTConsense WAL Interval Reading Strategy Optimization

### PR DESCRIPTION
**Pull request by BUAAserein**
#### Description

IoTConsensus WAL always resets the insertNodes when WAL skip to a new targetIndex. But in most writting process, targetIndex is sequentially increasing, which means we may use the previous insertNodes as a cache and doesn't need to reset it everytime.

### CODE REVIEW

**Review by caozj1011 — APPROVED** [review]

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/node/WALNode.java [hunk 1]
```diff
@@ -833,6 +833,21 @@ public void skipTo(long targetIndex) {
             targetIndex,
             targetIndex);
       }
+
+      if (itr != null
+          && itr.hasNext()
+          && insertNodes.get(itr.nextIndex()).getSearchIndex() <= targetIndex
+          && targetIndex <= insertNodes.get(insertNodes.size() - 1).getSearchIndex()) {
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/wal/node/WALNode.java:840]: How about changing the type of insertNodes to LinkedList, so you can use `insertNodes.getLast().getSearchIndex()` here

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM~
Please do some performance testing and post the result to this pr~

**Comment by BUAAserein** [comment]
After testing the long-term and short-term performance, we found that the write performance decreased while the sync performance increased, which means that the sync performance was improved. Writing performance dropped by approximately 0.61%, and synchronization performance increased by approximately 0.29%.

**Comment by OneSizeFitsQuorum** [comment]
see [doc](https://apache-iotdb.feishu.cn/docx/YJDldH3eoo2hffx70oxcuE1XnUg) for details

**Pull request merged**